keras_nlp/models/xlnet/xlnet_content_and_query_embedding.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""XLNet Content and Query Embedding implementation"""
+
+
+from keras_nlp.backend import keras
+from keras_nlp.backend import ops
+
+
+class ContentAndQueryEmbedding(keras.layers.Layer):
+    """
+    Content and Query Embedding.
+
+    This class creates Content and Query Embeddings for XLNet model
+    which is later used in XLNet ENcoder.
+
+    Args:
+        vocabulary_size: int, number of tokens in the vocabulary.
+        hidden_dim: int, the size hidden states.
+        dropout: float, defaults to 0. the dropout value, shared by
+            `keras.layers.TwoStreamRelativeAttention` and feedforward network.
+        kernel_initializer_range: int, defaults to 0.02. The kernel initializer
+            range for the dense and relative attention layers.
+        name: string, defaults to None. The name of the layer.
+        **kwargs: other keyword arguments.
+
+    References:
+     - [XLNet: Generalized Autoregressive Pretraining for Language Understanding]
+     (https://arxiv.org/abs/1906.08237)
+    """
+
+    def __init__(
+        self, vocabulary_size, hidden_dim, dropout, name=None, **kwargs
+    ):
+        super().__init__(name=name, **kwargs)
+        self.vocabulary_size = vocabulary_size
+        self.hidden_dim = hidden_dim
+        self.dropout = dropout
+
+        self._built = None
+
+    def positional_embedding(self, pos_seq, inv_freq, bsz=None):
+        sinusoid_inp = ops.einsum("i,d->id", pos_seq, inv_freq)
+        pos_emb = ops.concatenate(
+            [ops.sin(sinusoid_inp), ops.cos(sinusoid_inp)], axis=-1
+        )
+        pos_emb = pos_emb[:, None, :]
+
+        if bsz is not None:
+            import tensorflow as tf
+
+            pos_emb = tf.tile(pos_emb, [1, bsz, 1])
```

Comment:
I was getting errors while using `ops.tile`, so I kept it for now. I am currently working on to fix the error.

Reply:
I have fixed it.